Mage.Sets/src/mage/cards/r/RinAndSeriInseparable.java

```diff
@@ -0,0 +1,119 @@
+        catSpellFilter.add(SubType.CAT.getPredicate());
+
+        dogPermanentFilter.add(SubType.DOG.getPredicate());
+        dogPermanentFilter.add(new ControllerIsActivePlayerPredicate());
+        catPermanentFilter.add(SubType.CAT.getPredicate());
+        catPermanentFilter.add(new ControllerIsActivePlayerPredicate());
+    }
+
+    public RinAndSeriInseparable(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}{G}{W}");
+        
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.DOG);
+        this.subtype.add(SubType.CAT);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Whenever you cast a Dog spell, create a 1/1 green Cat creature token.
+        this.addAbility(new SpellCastControllerTriggeredAbility(
+                new CreateTokenEffect(new CatToken3()), dogSpellFilter, false
+        ));
+
+        // Whenever you cast a Cat spell, create a 1/1 white Dog creature token.
+        this.addAbility(new SpellCastControllerTriggeredAbility(
+                new CreateTokenEffect(new DogToken2()), catSpellFilter, false
+        ));
+
+        // {R}{G}{W}: Rin and Seri, Inseparable deals damage to any target equal to the number of Dogs you control. You gain life equal to the number of Cats you control.
+        Effect damageEffect = new DamageTargetEffect(new DynamicValue() {
+            @Override
+            public int calculate(Game game, Ability sourceAbility, Effect effect) {
+                return game.getBattlefield().count(dogPermanentFilter, sourceAbility.getSourceId(), sourceAbility.getControllerId(), game);
+            }
+
+            @Override
+            public DynamicValue copy() {
+                return this;
+            }
+
+            @Override
+            public String getMessage() {
+                return "Dogs you control";
+            }
+        });
+        damageEffect.setText("{source} deals damage to any target equal to the number of Dogs you control");
+        Effect lifeGainEffect = new GainLifeEffect(new DynamicValue() {
+            @Override
+            public int calculate(Game game, Ability sourceAbility, Effect effect) {
+                return game.getBattlefield().count(catPermanentFilter, sourceAbility.getSourceId(), sourceAbility.getControllerId(), game);
+            }
+
+            @Override
+            public DynamicValue copy() {
+                return this;
+            }
+
+            @Override
+            public String getMessage() {
+                return "Cats you control";
+            }
+        });
+        lifeGainEffect.setText("You gain life equal to the number of Cats you control.");
+        Ability ability = new SimpleActivatedAbility(damageEffect, new ManaCostsImpl("{R}{G}{W}"));
+        ability.addTarget(new TargetAnyTarget());
+        ability.addEffect(lifeGainEffect);
```

Comment:
Try to use xmage's standard order code: ability -> effect -> target (that's order uses in most cases).

Reply:
fixed